jetbrains-core/src/software/aws/toolkits/jetbrains/utils/CoroutineDispatchers.kt

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Runnable
+import kotlin.coroutines.CoroutineContext
+
+object CoroutineDispatchers {
+    fun dispatcherFor(modalityState: ModalityState): CoroutineDispatcher =
```

Comment:
What's the intend here? Are there other places we should be using this?

Reply:
intent here was to see what this would look like before committing to converting our other usages of `runInEdt` to this